table-api/src/main/java/io/deephaven/api/filter/ExtractAnds.java

```diff
@@ -60,6 +59,16 @@ public Collection<Filter> visit(FilterSerial serial) {
         return Collections.singleton(serial);
     }
 
+    @Override
+    public Collection<Filter> visit(FilterBarrier barrier) {
+        return Collections.singleton(barrier);
```

Comment:
Is it appropriate to descend here?

Reply:
We can't descend on barrier; because we can't duplicate the barrier.  We should probably descend on respecting the barrier (and in this method tell you why).